### PR TITLE
Store question answers

### PR DESCRIPTION
**Pull request by oddkk**

### CODE REVIEW

**Comment by oddkk** [comment]
This pull request resolves issue #23

File: src/question/asker.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+        if error:
+            return error, None
+
+        error, answer_id = self.answer_repository.create(
+                question.id, user_id
+        )
+
+        if error:
+            return error, None
```
Comment (michaelmcmillan) [inline src/question/asker.py]: Perhaps combine this if-statement with the one above using something like ````question_error and answer_error````?

File: src/question/asker.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    def answer(self, user_id, answer_id, data):
+        try:
+            user_id = int(user_id)
+        except ValueError:
+            return QuestionAnswerRepository.ANSWER_DOES_NOT_EXIST, None
+        try:
+            answer_id = int(answer_id)
+        except ValueError:
+            return QuestionAnswerRepository.ANSWER_DOES_NOT_EXIST, None
```
Comment (michaelmcmillan) [inline src/question/asker.py]: Maybe group these try-catches as well?

File: src/question/question_answer_repository.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    def __init__(self, database):
+        self.database = database
+
+    def create(self, question, user):
+        row = {'question': question, 'user': user}
+        result = self.database.execute(self.CREATE, row)
+        return (None, result.lastrowid)
+
+    def answer(self, answer_id, answer, correct):
+        row = {
```
Comment (michaelmcmillan) [inline src/question/question_answer_repository.py]: I suggest extracting an inverse function of `from_row_to_question_answer` named `from_question_answer_to_row` :)

File: src/webserver/controllers/question.py [hunk 4]
```diff
@@ -1,27 +1,71 @@
...
-        error, question = question_repository.get_by_id(question)
+    @login_required
+    def next_question_from_topic(cls, topic):
+        asker = Asker(g.db)
+
+        user_id = current_user.user.id
+
+        error, answer_id = asker.next_question_from_topic(user_id, topic)
+
+        if error:
```
Comment (michaelmcmillan) [inline src/webserver/controllers/question.py]: Redundant `if`?

File: src/question/asker.py (hunk 1 above)
Comment (oddkk) [inline src/question/asker.py]: If we do not find a question, we can not create an answer. We want to abort before attempting to create an answer in this case.

File: src/question/asker.py (hunk 2 above)
Comment (oddkk) [inline src/question/asker.py]: Ah, good idea :) Will fix

File: src/question/question_answer_repository.py (hunk 3 above)
Comment (oddkk) [inline src/question/question_answer_repository.py]: Good idea :) Will do

File: src/webserver/controllers/question.py (hunk 4 above)
Comment (oddkk) [inline src/webserver/controllers/question.py]: I can rewrite this a bit, but I want to make a distinction of what error occurred. Some of the errors should return 404, while the rest should return 500

**Pull request merged**